### DISCUSSION TITLE
Eco Charge/Discharge to native integration

### DISCUSSION THREAD

**Discussion by ahammas** [mletenay/home-assistant-goodwe-inverter#247] [Q&A]
Hi!

I'm clumsy when fiddling around with custom components and I'm afraid I'll screw things up (lose historic data) by installing the experimental version of the GW integration in my home assistant. The native version has almost everything I need so I'm very hesitant to risk it by going experimental. What I'm lacking in the native version is a good way of controlling charge/discharge time. So I'm wondering when this function might be making it to the native version? I think this would be the single greatest improvement of an already great integration.

Cheers
Anders

**Comment by ahammas** [comment]
I jumped into the deep end (impatient) and installed the experimental version. It worked like a charm. Only "issue" I see is that I have two "sync inverter clock" entities where one doesn't work.

**Comment by AlexPanoptikum** [comment]
@ahammas 
Hallo,
I more or less have the same problem, can you please tell me how you changed from to the experimental version. 
Did you first disable the regular version or delete it? And you really didn't loose any data? Last question, do you know if this works with the EM series as well?

I allready had built a nice solution, I think, it just does not work, because I can' start and stop properly with the regular integration.

so here is my setting (Readme.md made with help of chatgpt):
Attention again, it doesn't work because of missing modes to start and stop charging!

[README.md](https://github.com/user-attachments/files/20458202/README.md)
